packages/core-transaction-pool/lib/interface.js

```diff
@@ -280,6 +280,21 @@ module.exports = class TransactionPoolInterface {
     this.walletManager.deleteWallet(senderPublicKey)
   }
 
+  /**
+   * Purges all transactions from senders with at least one
+   * invalid transaction.
+   * @param {Block} block
+   */
+  purgeSendersWithInvalidTransactions(block) {
+    const publicKeys = new Set(
+      block.transactions
+        .filter(tx => !tx.verified)
```

Comment:
Isn't it better to prevent transactions with `.verified == false` from entering the pool in the first place, rather than letting them in and removing them afterwards?

Reply:
this is done in the guard